appengine-java8/websocket-jetty/src/main/java/com/example/flexible/websocket/jettynative/SendServlet.java

```diff
@@ -73,22 +77,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
   }
 
-  private WebSocketClient createWebSocketClient() {
-    WebSocketClient webSocketClient;
+  private HttpClient createHttpClient() {
+    HttpClient httpClient;
     if (System.getenv(GAE_INSTANCE_VAR) != null) {
       // If on HTTPS, create client with SSL Context
-      SslContextFactory sslContextFactory = new SimpleContainerScope(
-              WebSocketPolicy.newClientPolicy())
-              .getSslContextFactory();
-      webSocketClient = new WebSocketClient(sslContextFactory);
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      httpClient = new HttpClient(sslContextFactory);
     } else {
       // local testing on HTTP
-      webSocketClient = new WebSocketClient();
+      httpClient = new HttpClient();
     }
-    return webSocketClient;
+    return httpClient;
+  }
+
+  private WebSocketClient createWebSocketClient() {
+    return new WebSocketClient(this.httpClient);
   }
 
   private void sendMessageOverWebSocket(String message) throws Exception {
+    if (!httpClient.isRunning()) {
+      try {
+        httpClient.start();
+      } catch (URISyntaxException e) {
+        e.printStackTrace();
```

Comment:
I know this is consistent with `webSocketClient`, but I don't think just printing the stack trace is the best idea.

Reply:
Yup, I agree.
My personal opinion is to treat `.start()` exceptions as fatal and prevent further use.